File: eventmanagementsystem_ui/src/services/AttendeeService.js
import http from "../http-common";

class AttendeeService {
  getAllAttendees(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/attendee/attendees`, searchDTO);
  }

  get(attendeeId) {
    return this.getRequest(`/attendee/${attendeeId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/attendee?field=${matchData}`, null);
  }

  addAttendee(data) {
    return http.post("/attendee/addAttendee", data);
  }

  update(data) {
  	return http.post("/attendee/updateAttendee", data);
  }
  
  uploadImage(data,attendeeId) {
  	return http.postForm("/attendee/uploadImage/"+attendeeId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AttendeeService();

File: src/main/java/com/eventmanagementsystem/service/AttendeeService.java
package com.eventmanagementsystem.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.eventmanagementsystem.domain.Attendee;
import com.eventmanagementsystem.dto.AttendeeDTO;
import com.eventmanagementsystem.dto.AttendeeSearchDTO;
import com.eventmanagementsystem.dto.AttendeePageDTO;
import com.eventmanagementsystem.dto.AttendeeConvertCriteriaDTO;
import com.eventmanagementsystem.service.GenericService;
import com.eventmanagementsystem.dto.common.RequestDTO;
import com.eventmanagementsystem.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AttendeeService extends GenericService<Attendee, Integer> {

	List<Attendee> findAll();

	ResultDTO addAttendee(AttendeeDTO attendeeDTO, RequestDTO requestDTO);

	ResultDTO updateAttendee(AttendeeDTO attendeeDTO, RequestDTO requestDTO);

    Page<Attendee> getAllAttendees(Pageable pageable);

    Page<Attendee> getAllAttendees(Specification<Attendee> spec, Pageable pageable);

	ResponseEntity<AttendeePageDTO> getAttendees(AttendeeSearchDTO attendeeSearchDTO);
	
	List<AttendeeDTO> convertAttendeesToAttendeeDTOs(List<Attendee> attendees, AttendeeConvertCriteriaDTO convertCriteria);

	AttendeeDTO getAttendeeDTOById(Integer attendeeId);







}





